shell/platform/darwin/common/framework/Source/FlutterChannels.mm

```diff
@@ -9,12 +9,28 @@
 #pragma mark - Basic message channel
 
 static NSString* const kFlutterChannelBuffersChannel = @"dev.flutter/channel-buffers";
+static NSString* const kResizeMethod = @"resize";
+static NSString* const kOverflowMethod = @"overflow";
 
 static void ResizeChannelBuffer(NSObject<FlutterBinaryMessenger>* binaryMessenger,
                                 NSString* channel,
                                 NSInteger newSize) {
-  NSString* messageString = [NSString stringWithFormat:@"resize\r%@\r%@", channel, @(newSize)];
-  NSData* message = [messageString dataUsingEncoding:NSUTF8StringEncoding];
+  NSArray* args = @[ channel, [NSNumber numberWithInt:newSize] ];
```

Comment:
Given that NSInteger supports negative values, we should
```cc
NSAssert(newSize >= 0, @"Channel buffer size must be non-negative");
```
at the top of this method to help catch bugs.